### PR TITLE
Update ecl to add ability to recover from bad magnetic yaw

### PR DESCRIPTION
**Pull request by priseborough**
Required to log additional data required for flight testing when ecl version updates to latest master containing https://github.com/PX4/ecl/commit/4669aa63121c6f91e2e034b6da05e35fbba98ca1

### CODE REVIEW

File: msg/vehicle_status_flags.msg [hunk 1]
```diff
@@ -36,3 +36,5 @@ bool usb_connected                                # status of the USB power supp
 
 bool avoidance_system_required					  # Set to true if avoidance system is enabled via COM_OBS_AVOID parameter
 bool avoidance_system_valid                       # Status of the obstacle avoidance system
+
+uint8 emergency_yaw_reset_counter	# increments when the commander is requesting the EKF perfomr and emergencey yaw reset
```
Comment (dagar) [inline msg/vehicle_status_flags.msg]: I don't understand the intended interaction here. Is commander going to be setting this later?

**Comment by dagar** [comment]
Huge flash overflow on fmu-v2. Looks like it might be time for something a little more drastic...

File: msg/vehicle_status_flags.msg (hunk 1 above)
Comment (priseborough) [reply msg/vehicle_status_flags.msg]: Have removed this. It was a hangover from an earlier design.

File: msg/CMakeLists.txt [hunk 2]
```diff
@@ -162,7 +162,8 @@ set(msg_files
 	vtol_vehicle_status.msg
 	wheel_encoders.msg
 	wind_estimate.msg
-)
+	yaw_est_test_data.msg
```
Comment (bresch) [inline msg/CMakeLists.txt]:
```suggestion
	yaw_estimator_status.msg or yaw_estimate.msg ?
```

File: src/modules/ekf2/ekf2_main.cpp [hunk 3]
```diff
@@ -525,7 +528,11 @@ class Ekf2 final : public ModuleBase<Ekf2>, public ModuleParams, public px4::Sch
 		_param_ekf2_move_test,	///< scaling applied to IMU data thresholds used to determine if the vehicle is static or moving.
 
 		(ParamFloat<px4::params::EKF2_REQ_GPS_H>) _param_ekf2_req_gps_h, ///< Required GPS health time
-		(ParamExtInt<px4::params::EKF2_MAG_CHECK>) _param_ekf2_mag_check ///< Mag field strength check
+		(ParamExtInt<px4::params::EKF2_MAG_CHECK>) _param_ekf2_mag_check, ///< Mag field strength check
+
+		// Used by EKF-GSF experimental yaw estimator
+		(ParamExtFloat<px4::params::EKF2_GSF_TAS>)
+		_param_ekfgsf_tas_default	///< default value of true airspeed assumed during fixed wing operation
```
Comment (bresch) [inline src/modules/ekf2/ekf2_main.cpp]:
```suggestion
		_param_ekf2_gsf_tas_default	///< default value of true airspeed assumed during fixed wing operation
```

File: src/modules/ekf2/ekf2_main.cpp [hunk 4]
```diff
@@ -1597,6 +1605,8 @@ void Ekf2::Run()
 
 			publish_wind_estimate(now);
 
+			publish_test_data(now);
```
Comment (bresch) [inline src/modules/ekf2/ekf2_main.cpp]:
```suggestion
			publish_yaw_estimator_status(now);
```

File: src/modules/ekf2/ekf2_main.cpp [hunk 5]
```diff
@@ -1774,6 +1784,26 @@ bool Ekf2::publish_attitude(const hrt_abstime &now)
...
+			       &yaw_est_test_data.yaw[0],
+			       &yaw_est_test_data.innov_vn[0], &yaw_est_test_data.innov_ve[0],
+			       &yaw_est_test_data.weight[0])) {
+
+		yaw_est_test_data.timestamp = timestamp;
+
+		_yaw_est_pub.publish(yaw_est_test_data);
+
+		return true;
+
```
Comment (bresch) [inline src/modules/ekf2/ekf2_main.cpp]:
```suggestion
```

File: src/modules/logger/logger.h [hunk 6]
```diff
@@ -55,8 +55,11 @@
 
 extern "C" __EXPORT int logger_main(int argc, char *argv[]);
 
-static constexpr hrt_abstime TRY_SUBSCRIBE_INTERVAL{1000 * 1000};	// interval in microseconds at which we try to subscribe to a topic
-// if we haven't succeeded before
+#ifdef ORB_USE_PUBLISHER_RULES
+static constexpr hrt_abstime TRY_SUBSCRIBE_INTERVAL {1000};
+#else
+static constexpr hrt_abstime TRY_SUBSCRIBE_INTERVAL {1000 * 1000}; // interval in microseconds at which we try to subscribe to a topic
+#endif
```
Comment (bresch) [inline src/modules/logger/logger.h]: Why do we need that?

File: msg/CMakeLists.txt (hunk 2 above)
Comment (dagar) [reply msg/CMakeLists.txt]: Adding an `ekf2_` prefix might also make sense.

File: msg/yaw_est_test_data.msg [hunk 7]
```diff
@@ -0,0 +1,7 @@
+uint64 timestamp	# time since system start (microseconds)
+float32 yaw_composite	# composite yaw from GSF (rad)
+float32 yaw_variance	# composite yaw variance from GSF (rad^2)
+float32[8] yaw		# yaw estimate for each model in the filter bank (rad)
```
Comment (dagar) [inline msg/yaw_est_test_data.msg]:
```suggestion
float32[5] yaw		# yaw estimate for each model in the filter bank (rad)
```

**Comment by dagar** [comment]
Fix for CI clang-tidy failure https://github.com/PX4/ecl/pull/775.

File: msg/CMakeLists.txt (hunk 2 above)
Comment (dagar) [reply msg/CMakeLists.txt]: Renamed.

File: src/modules/ekf2/ekf2_main.cpp (hunk 3 above)
Comment (dagar) [reply src/modules/ekf2/ekf2_main.cpp]: Done.

File: src/modules/ekf2/ekf2_main.cpp (hunk 4 above)
Comment (dagar) [reply src/modules/ekf2/ekf2_main.cpp]: Done.

File: src/modules/logger/logger.h (hunk 6 above)
Comment (dagar) [reply src/modules/logger/logger.h]: Removed for now. Might be less relevant after https://github.com/PX4/Firmware/pull/14106. I'll readd this (or solve it differently) if needed in master.

File: msg/yaw_est_test_data.msg (hunk 7 above)
Comment (dagar) [reply msg/yaw_est_test_data.msg]: Size fixed.

**Comment by jorge789** [comment]
### Tested on PixRacer V4

**Indoor Flight Test**
Modes Tested
Stabilized Mode: Good.

**Procedure**
Arm and Take off in stabilized mode.

**Notes**
No issues noted, good flight in general.

**Log**  https://review.px4.io/plot_app?log=e5d5c88a-6cd0-4297-880c-54328d4c2018

### Tested on CUAV nano V5

**Indoor Flight Test**
Modes Tested
Stabilized Mode: Good.

**Procedure**
Arm and Take off in stabilized mode.

**Notes**
No issues noted, good flight in general.

**Log**  https://review.px4.io/plot_app?log=442d5bf6-b8ac-48bc-b8dc-7befe8525b27

**Pull request merged**